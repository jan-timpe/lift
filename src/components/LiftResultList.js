import React, { Component } from 'react'
import LiftResult from './LiftResult'


export default class LiftResultList extends Component {

  render() {
    return (
      <div className="LiftResultList">
        {this.props.results.map((result, index) => (
          <LiftResult result={result} key={index} />
        ))}
      </div>
    )
  }

}