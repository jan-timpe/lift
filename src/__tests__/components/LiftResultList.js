import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LiftResultList from '../../components/LiftResultList'

describe(`LiftResultList`, () => {

  // snapshot
  it(`should render`, () => {
    const results = [{
      id: 0,
      liftId: 0,
      weight: 200,
      date: '7/1/2018'
    }, {
      id: 1,
      liftId: 0,
      weight: 300,
      date: '7/2/2018'
    }]
    const wrapper = shallow(<LiftResultList results={results} />)
    const component = wrapper
    expect(toJson(component)).toMatchSnapshot()
  })

})