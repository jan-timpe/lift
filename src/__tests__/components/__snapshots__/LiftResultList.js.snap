// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiftResultList should render 1`] = `
<div
  className="LiftResultList"
>
  <LiftResult
    key="0"
    result={
      Object {
        "date": "7/1/2018",
        "id": 0,
        "liftId": 0,
        "weight": 200,
      }
    }
  />
  <LiftResult
    key="1"
    result={
      Object {
        "date": "7/2/2018",
        "id": 1,
        "liftId": 0,
        "weight": 300,
      }
    }
  />
</div>
`;
